crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -109,8 +114,66 @@ impl Plane3d {
     /// are *collinear* and lie on the same line.
     #[inline(always)]
     pub fn from_points(a: Vec3, b: Vec3, c: Vec3) -> (Self, Vec3) {
-        let normal = Dir3::new((b - a).cross(c - a))
-            .expect("plane must be defined by three finite points that don't lie on the same line");
+        let normal = Dir3::new((b - a).cross(c - a)).expect(
+            "finite plane must be defined by three finite points that don't lie on the same line",
+        );
+        let translation = (a + b + c) / 3.0;
+
+        (
+            Self {
+                normal,
+                ..Default::default()
+            },
+            translation,
+        )
+    }
+}
+
+/// An unbounded plane in 3D space. It forms a separating surface through the origin,
+/// stretching infinitely far
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+pub struct InfinitePlane3d {
+    /// The normal of the plane. The plane will be placed perpendicular to this direction
+    pub normal: Dir3,
+}
+impl Primitive3d for InfinitePlane3d {}
+
+impl Default for InfinitePlane3d {
+    /// Returns the default [`InfinitePlane3d`] with a normal pointing in the `+Y` direction.
+    fn default() -> Self {
+        Self { normal: Dir3::Y }
+    }
+}
+
+impl InfinitePlane3d {
+    /// Create a new `InfinitePlane3d` from a normal
+    ///
+    /// # Panics
+    ///
+    /// Panics if the given `normal` is zero (or very close to zero), or non-finite.
+    #[inline(always)]
+    pub fn new(normal: Vec3) -> Self {
+        Self {
+            normal: Dir3::new(normal).expect("normal must be nonzero and finite"),
+        }
+    }
```

Comment:
```suggestion
    pub fn new(normal: TryInto<Dir3>) -> Self {
        Self {
            normal: normal.try_into().expect("normal must be nonzero and finite"),
        }
    }
```

Reply:
Sorry, the suggested change here produces an error, I am curious what is the driver behind this suggestion?

I think other methods also expect a Vec3 so I wouldn't change just one's signature.

Reply:
If you have a `Dir3` this will do a redundant conversion to and from a `Vec`. It looks to me like this is the only place a vector is taken as a parameter and immediately converted into a `Dir3`.

It's fine either way. What's the error?

Reply:
Ah, it should be `impl TryInto<Dir3>`. Well anyway it's not super important.

Reply:
```suggestion
    pub fn new(normal: impl TryInto<Dir3>) -> Self {
        Self {
            normal: Dir3::new(normal).expect("normal must be nonzero and finite"),
        }
    }
```

Reply:
@NthTensor Sorry, I missed your comments here. Thank you for the explanation!

Reply:
@alice-i-cecile  This is now addressed!